test/README.md

```diff
@@ -0,0 +1,49 @@
+# How to run the end-to-end tests
+
+## Test parameters
+
+The end-to-end tests take several parameters, which they expect to find in a file named `context.json` in the same directory as the test binaries (e.g., `test/Microsoft.Azure.Devices.Edge.Test/bin/Debug/netcoreapp2.1/context.json`). Parameter names are case-insensitive. The parameters are:
+
+| Name | Required | Description |
+|------|----------|-------------|
+| `caCertScriptPath` | ✔ | Path to the folder containing `certGen.sh` (Linux) or `ca-certs.ps1` (Windows). |
+| `deviceId` || Name by which the edge device-under-test will be registered in IoT Hub. If not given, a name will automatically be generated in the format: `e2e-{device hostname}-{yyMMdd-HHmmss.fff}`. |
+| `edgeAgentImage` || Docker image to pull/use for Edge Agent. Regardless of what you put here, the default `mcr.microsoft.com/azureiotedge-agent:1.0` will be used in config.yaml to start IoT Edge; this setting only applies to any configurations deployed by the tests. |
+| `edgeHubImage` || Docker image to pull/use for Edge Hub. |
+| `installerPath` || Path to the Windows installer script `IotEdgeSecurityDaemon.ps1`. This parameter is ignored on Linux, and optional on Windows. If not given on Windows, the default script will be downloaded from https://aka.ms/iotedge-win to a temporary location. |
+| `logFile` || Path to which all test output will be written, including verbose output. This setting allows the user to capture all the details of a test pass while keeping the shell window output free of visual clutter. Note that daemon logs and module logs are always written to the same directory as the test binaries (e.g., `test/Microsoft.Azure.Devices.Edge.Test/bin/Debug/netcoreapp2.1/*.log`), independent of this parameter. |
+| `methodReceiverImage` || Docker image to pull/use for the 'DirectMethodReceiver' module (see the test 'ModuleToModuleDirectMethod'). |
+| `methodSenderImage` || Docker image to pull/use for the 'DirectMethodSender' module (see the test 'ModuleToModuleDirectMethod'). |
+| `optimizeForPerformance` || Boolean value passed to Edge Hub. Usually set to 'false' when running on more constrained platforms like Raspberry Pi. If not given, it defaults to 'true'. |
```

Comment:
Should we determine this value based on OS platform and remove this parameter?

Reply:
Sure, but different PR please. :)